### ISSUE TITLE
GATED research: evidence consistency, calibration, and abstention

### ISSUE THREAD

**Issue by gr-rr** [ListenCloser/listencloser#1309]
Program owner: #1303.
Research synthesis: #1297.
Relevant capability owners include #931 (general transcription), #1233 (Melody), #1236/#1087 (Pitch), #1194 (Harmony), #1176 (Layers), #1181 (lyrics alignment), #1083 (Score↔performance alignment), #613 (representation authority).

## Status

**GATED SECOND WAVE — do not start model training until there is enough overlapping evidence plus defensible correctness/failure labels.**

## Question

> **Can a small calibrated model learn when ListenCloser should trust, route, fuse, or abstain across overlapping evidence sources better than fixed engine-specific thresholds?**

This is not a new detector. It is a decision layer over already-produced evidence.

## Motivation

ListenCloser increasingly has multiple views of the same musical event or passage:

```text
audio transcription
symbolic Melody
continuous F0
source/performance Score alignment
chords
beats
stems
lyrics alignment
alternate engines
```

Disagreement itself may be useful evidence. Example:
- transcription + aligned Score + F0 agree on a note → stronger trust;
- transcription disagrees with F0 and has no score match → candidate abstention/correction request;
- a separated vocal may be sufficient for lyrics timing but unreliable for pitch, so stem quality should be task-specific rather than one generic SDR-like confidence.

The proprietary hypothesis is that **LC can learn evidence reliability/routing without retraining base MIR systems**.

## Gate to start

Do not fit a model until at least one bounded decision problem has all of:

- [ ] two or more overlapping evidence sources on the same exact source/Version/time domain;
- [ ] legitimate correctness/reference labels or a defensible task outcome;
- [ ] enough positive + failure examples to evaluate calibration, not only average accuracy;
- [ ] deterministic authority/lineage mapping so source disagreement is not caused by mismatched Versions;
- [ ] a fixed rule-based baseline to beat.

Preferred first task should be whichever naturally accumulates the cleanest labels; examples:
- note-level transcription confidence using exact Score/performance alignment + F0 on aligned piano material;
- lyrics-alignment input/router choice using mixture vs vocal stem where reference timings exist;
- task-specific stem reliability for a downstream pitch/lyrics metric;
- choosing between alternate symbolic/audio interpretations when reference labels exist.

Do not create a universal cross-capability confidence model in v0.

## Data contract

A row should be bound to exact source/event identity and include only information available at decision time:

```text
source/version identity
domain/task
engine/source outputs
engine-native confidence where semantically valid
disagreement features
coverage/missingness
simple signal-quality features if legitimate
reference correctness / downstream-success label
```

Do not turn model-native confidence into generic truth confidence without calibration.

## Baselines

Compare against simple decision rules first:

```text
A. trust primary engine always
B. fixed threshold on primary engine-native confidence where available
C. simple agreement/disagreement rule
D. calibrated logistic regression / small tree model
```

Only use a tiny MLP if the simpler calibrated models clearly underfit.

## Calibration methods

Prefer established small-data methods:
- logistic regression;
- Platt scaling;
- isotonic regression when enough data exists;
- temperature scaling if calibrating logits from one model;
- simple gradient-boosted trees for heterogeneous evidence features if needed.

No large neural fusion network initially.

## Metrics

Primary metric depends on the chosen decision task but must include **calibration/selective prediction**, not only classification accuracy.

Preferred metrics:
- AUROC/AUPRC for failure/correctness prediction where appropriate;
- Brier score / log loss;
- Expected Calibration Error with caveats and reliability diagrams;
- risk-coverage curve / selective accuracy as abstention increases;
- downstream metric after routing/fusion vs baseline.

The key product question is:

> At a chosen coverage level, can LC make fewer confident mistakes than the existing fixed policy?

## Compute budget

Expected CPU-only.

```text
GPU-hours = 0 preferred
cash cost = $0
```

Use scikit-learn / NumPy/Pandas. Any expensive upstream inference needed to materialize evidence should reuse existing evaluation/runtime paths rather than becoming training infrastructure.

## Success criterion

A useful result should demonstrate on held-out Works/sources that a tiny decision model:

- improves calibration/failure prediction over engine-native/fixed-rule baselines;
- yields a better risk-coverage tradeoff;
- or routes to alternate evidence in a way that improves the actual downstream task metric.

A model that merely learns genre/engine identity without predicting failures is not useful.

## Stop / kill criteria

Stop if:
- labels are too sparse/noisy to evaluate calibration honestly;
- overlapping evidence is not actually aligned to exact source identity;
- simple agreement rules perform as well as learned calibration;
- gains rely on leakage from post-hoc/reference-only features;
- the model needs a broad cross-capability feature schema/framework before one bounded task is proven.

## Product consequence if successful

A successful bounded model may justify a focused route such as:

```text
use primary result
use alternate result
show both as interpretations
abstain / qualify
request correction
```

It must never silently overwrite canonical source evidence.

## Non-goals

- no universal “ListenCloser confidence score”;
- no retraining base MIR models;
- no large multimodal fusion network;
- no generic router framework;
- no user-facing confidence percentage unless independently justified/calibrated;
- no replacement of #613 authority semantics.

## Deliverables

1. One bounded task chosen with exact evidence/label contract.
2. Frozen held-out source/Work split.
3. Fixed-rule and engine-native baselines.
4. Small calibrated-model comparison.
5. Reliability/risk-coverage plots or equivalent numeric artifacts.
6. Downstream routing/abstention impact where measurable.
7. Explicit disposition:

```text
FAIL — learned calibration adds no value over simple rules
NARROW — useful for one capability only
CONTINUE — strong held-out selective-prediction/routing gain; open focused integration owner
```

## Acceptance

- [ ] start gates satisfied for one bounded task;
- [ ] exact source/Version identity preserved;
- [ ] held-out calibration/selective metrics reported;
- [ ] simple rule baselines included;
- [ ] CPU/$0 target maintained;
- [ ] no universal-confidence abstraction introduced;
- [ ] disposition written back to #1303.

Refs #1303 #1297 #931 #1233 #1236 #1087 #1194 #1176 #1181 #1083 #613.

**Comment by gr-rr** [comment]
## Data-constrained sequencing update

This issue should remain gated on **one exact objective task**, but it does not need to wait behind #1307/#1308. Under the program's no-manual-label assumption, #1309 is the most plausible second-wave experiment to advance early because public benchmarks can supply correctness/failure labels.

### Preferred activation rule

Start as soon as one existing evaluation path provides all of:

```text
public/reference ground truth
>=2 overlapping evidence sources on the same exact source/time domain
enough successes + failures for calibration/selective prediction
fixed rule baseline
source/Work-disjoint evaluation
```

Candidate directions should be chosen from data already available to LC, not by starting a new acquisition/annotation program. Strong examples include:
- aligned symbolic/performance material where transcription + F0 + reference notes overlap;
- a public lyrics/alignment corpus once #1181 has two legitimate input/engine routes;
- any existing engine bakeoff with exact held-out truth and per-example disagreement outputs.

### LLM policy

Do not use an LLM to label whether an MIR event is correct when exact benchmark truth is unavailable. LLMs may summarize error clusters or propose features after evaluation, but must not manufacture held-out calibration truth.

### Compute

The calibrated model itself remains CPU/$0. Expensive upstream inference should be materialized once through existing evaluators and cached with exact engine/checkpoint/source identity; do not create a fusion-training platform.

If a qualifying public/reference task appears while #1304–#1306 run, this issue may proceed in parallel.

Refs #1303 #1304 #1306 #1311 #931 #1233 #1083 #1181.
